bciers/apps/compliance/src/app/components/compliance-summary/manage-obligation/apply-compliance-units/ApplyComplianceUnitsComponent.tsx

```diff
@@ -146,63 +236,84 @@ export default function ApplyComplianceUnitsComponent({
     return "BELOW";
   }, [formData, initialOutstandingBalance]);
 
-  const canSubmit = useMemo(() => {
+  // Check if Apply button should be enabled
+  const canApply = useMemo(() => {
+    const data = formData as ApplyComplianceUnitsFormData;
     return !!(
-      (formData as ApplyComplianceUnitsFormData)?.bccr_holding_account_id &&
-      (formData as ApplyComplianceUnitsFormData)
-        ?.total_quantity_to_be_applied &&
-      (formData as ApplyComplianceUnitsFormData)?.total_quantity_to_be_applied >
-        0 &&
+      shouldShowApplyButton &&
+      data?.total_quantity_to_be_applied &&
+      data?.total_quantity_to_be_applied > 0 &&
       complianceLimitStatus !== "EXCEEDS"
     );
-  }, [formData, complianceLimitStatus]);
+  }, [formData, shouldShowApplyButton, complianceLimitStatus]);
+
+  // Select the appropriate schema based on current phase
+  const currentSchema = useMemo(() => {
+    switch (currentPhase) {
+      case "confirmation":
+        return applyComplianceUnitsConfirmationSchema;
+      case "compliance_data":
+        return applyComplianceUnitsDataSchema;
+      default:
+        return applyComplianceUnitsBaseSchema;
+    }
+  }, [currentPhase]);
 
   return (
     <FormBase
-      readonly={isSubmitted}
-      schema={applyComplianceUnitsSchema}
+      readonly={status === "applied"}
+      schema={currentSchema}
       uiSchema={applyComplianceUnitsUiSchema}
       formData={formData}
       onChange={handleChange}
       onSubmit={handleSubmit}
       formContext={{
         reportingYear,
         chargeRate: (formData as ApplyComplianceUnitsFormData)?.charge_rate,
-        validateBccrAccount: (accountId: string) =>
-          getBccrComplianceUnitsAccountDetails(accountId, complianceSummaryId),
+        validateBccrAccount: getBccrAccountDetails,
         onValidAccountResolved: (response?: BccrComplianceAccountResponse) => {
-          if (response?.outstanding_balance) {
-            setInitialOutstandingBalance(response.outstanding_balance);
-          }
           setFormData((prev: ApplyComplianceUnitsFormData) => ({
             ...prev,
             ...response,
           }));
+          setCurrentPhase("confirmation");
         },
         onError: setErrors,
         complianceLimitStatus,
-        isSubmitted,
+        isApplied: status === "applied",
       }}
       className="w-full min-h-[62vh] flex flex-col justify-between"
     >
       <div>
-        {canSubmit && !isSubmitted && (
+        <FormAlerts errors={errors} />
+        {canApply && status !== "applied" && (
           <div className="mt-8">
             <ApplyComplianceUnitsAlertNote />
           </div>
         )}
-        <FormAlerts errors={errors} />
```

Comment:
I suggest keeping the error component here, just below the alert note, so it appears above the buttons if there are any.